game/players.py

```diff
@@ -21,25 +21,32 @@
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.distributions import Categorical
+import torch.optim as optim
 import numpy as np
 import random
-
+from types import SimpleNamespace
 
 class Player():
     '''
     A player class that will act like a template for the rest of the players/agents.
     This player is a random player that just chooses a random valid column to place its piece in
     '''
-    def __init__(self, _playerPiece, _winReward, _lossReward, _tieReward, _illegalReward, _device, _gamma = 0.8) -> None:
-        self.rewardValues = {"win": _winReward,
-                            "loss": _lossReward,
-                            "tie": _tieReward,
-                            "illegal": _illegalReward,
-                            "notEnded": 0}
-
-        self.playerPiece = _playerPiece
-        self.device = _device
-        self.gamma = _gamma
+    def __init__(self, player_piece: int, **kwargs) -> None:
+        self.params = {
+            "win_reward": 1,
+            "loss_reward": -1,
+            "tie_reward": 0.5,
+            "illegal_reward": -5,
+            "not_ended_reward": 0,
+            "reward_decay": 0.8,
+            "device": "cpu"
+        }
+        self.params.update(kwargs)  
+
+        self.playerPiece = player_piece
+        self.device = self.params["device"]
+        self.gamma = self.params["reward_decay"]
```

Comment:
Isn't it a bit confusing to call it `reward_decay` one place (in the arguments to `__init___`) and then assign it to `self.gamma`? Would be more consistent to stick with one name - we probably already have a few problems with inconsistency in our code 😆